source/Engine.cpp

```diff
@@ -2095,15 +2096,24 @@ void Engine::DoWeather(Weather &weather)
 		const Hazard *hazard = weather.GetHazard();
 		double multiplier = weather.DamageMultiplier();
 
-		// Get all ship bodies that are touching a ring defined by the hazard's min
-		// and max ranges at the hazard's origin. Any ship touching this ring takes
-		// hazard damage.
-		for(Body *body : shipCollisions.Ring(weather.Origin(), hazard->MinRange(), hazard->MaxRange()))
+		// System wide hazards do not act like an explosion, and do not have a limiting range.
+		if(hazard->SystemWide())
 		{
-			Ship *hit = reinterpret_cast<Ship *>(body);
-			double distanceTraveled = weather.Origin().Distance(hit->Position()) - hit->GetMask().Radius();
-			hit->TakeDamage(visuals, *hazard, multiplier, distanceTraveled, weather.Origin(), nullptr, hazard->BlastRadius() > 0.);
+			const System *system = player.GetSystem();
+			for(shared_ptr<Ship> ship: ships)
+				if(ship->GetSystem() == system)
+					ship->TakeDamage(visuals, *hazard, multiplier, 0., weather.Origin(), nullptr);
```

Comment:
This makes it so system-wide hazards don't scale with the distance of the ship to the hazard.

Why are ranged hazards and system-wide hazards treated differently anyways? Shouldn't system-wide hazards behave as-if we had an ranged hazard with an infinite range?

Reply:
The main use-case right now for system-wide hazards is things with a static effect no matter where one is in the system. 

It could be useful to allow infinite range-but-scaling in order to allow for, say, better simulated gravity, but not necessary for the core purpose here.

Reply:
This sounds like we need an "origin effect" and an "impact effect". Moving the origin of the hazard feels like the wrong solution. (Similar to weapons that have a fire-effect and an impact effect.)

Reply:
> This sounds like we need an "origin effect" and an "impact effect". Moving the origin of the hazard feels like the wrong solution. (Similar to weapons that have a fire-effect and an impact effect.)

We indeed have an origin and an impact effect, the impact effect is shown on every ship affected by the hazard (in this case, if system wide, all ships) and the origin effect is shown at the position of the flagship, aka the center of your screen so you can't miss it without needing to draw it on the entire system and lose performances.

Reply:
> This makes it so system-wide hazards don't scale with the distance of the ship to the hazard.
> 
> Why are ranged hazards and system-wide hazards treated differently anyways? Shouldn't system-wide hazards behave as-if we had a ranged hazard with an infinite range?

Infinite range? That seems very much undefined to me, the computer won't like that x)
Also, on a more serious note, you always could do what you seem to be talking about that by setting the max range for your hazard to 1 000 000 (but maybe I misunderstood)
This is, in fact, infinite range, it's just that it affects all ships in the same way because, as Zitchas stated it, it is what was needed in this case of ember waste hazards.